build/scripts/CrossBranchPorting.psm1

```diff
@@ -0,0 +1,147 @@
+
+    # Get the pull request details
+    $pullRequestDetails = (gh pr view $PullRequestNumber --json title,number,body,headRefName,baseRefName,mergeCommit,potentialMergeCommit | ConvertFrom-Json)
+    Write-Host "Backport to: $($TargetBranches -join ",")" -ForegroundColor Cyan
+    Write-Host "Pull Request Source Branch: $($pullRequestDetails.headRefName)" -ForegroundColor Cyan
+    Write-Host "Pull Request Target Branch: $($pullRequestDetails.baseRefName)" -ForegroundColor Cyan
+    Write-Host "Pull Request Title: $($pullRequestDetails.title)" -ForegroundColor Cyan
+    Write-Host "Pull Request Description: `n$($pullRequestDetails.body)" -ForegroundColor Cyan
+
+    if (-not $SkipConfirmation) {
+        GetConfirmation -Message "Please review the about information and press (y)es to continue or any other key to stop"
+    }
+
+    # Get the list of existing pull requests
+    $existingOpenPullRequests = gh pr list --state open --json title,state,baseRefName,url | ConvertFrom-Json
+
+    # Get the current branch before starting to backport
+    $startingBranch = RunAndCheck git rev-parse --abbrev-ref HEAD
+
+    $pullRequests = @{}
+    $branchNameSuffix = "$PullRequestNumber/$(Get-Date -Format "yyyyMMddHHmmss")"
+    foreach($TargetBranch in $TargetBranches) {
+        try {
+            $title = "[$TargetBranch] $($pullRequestDetails.title)"
+            $body = "This pull request backports #$($pullRequestDetails.number) to $TargetBranch"
+            $body += "`r`n`r`nFixes AB#[**Insert Work Item Number Here**]"
+
+            # Check if there is already a pull request for this branch
+            $existingPr = $existingOpenPullRequests | Where-Object { ($_.title -eq $title) -and ($_.baseRefName -eq $TargetBranch) }
+            if ($existingPr) {
+                Write-Host "Pull request for $TargetBranch already exists: $($existingPr.url)" -ForegroundColor Yellow
+                continue
+            }
+
+            # Create a new branch for the cherry-pick
+            $cherryPickBranch = "hotfix/$TargetBranch/$branchNameSuffix"
+
+            # Port the pull request to the target branch
+            PortPullRequest -PullRequestDetails $pullRequestDetails -TargetBranch $TargetBranch -CherryPickBranch $cherryPickBranch
+
+            # Create a pull request for the cherry-pick
+            gh pr create --title $title --body $body --base $TargetBranch --head $cherryPickBranch
+
+            $backportPr = gh pr view $cherryPickBranch --json url | ConvertFrom-Json
+            if ($backportPr.url) {
+                $pullRequests.Add($TargetBranch, $backportPr.url)
+            }
+        } catch {
+            RunAndCheck git checkout $startingBranch
+            Write-Host "Failed to backport to $TargetBranch. Please inspect the error and try again." -ForegroundColor Red
+            Write-Host $_.Exception.Message -ForegroundColor Red
+            throw $_.Exception.Message
```

Comment:
This would stop the whole loop. Wouldn't it better to continue trying to create the PRs for the other target branches?

Reply:
I'm not sure to be honest. If we fail to open a PR in one branch there's a (relatively high) risk it will fail for the other branches as well. It might just leave the developer with a bunch of new branches they have to clean up after and no backport PRs created🤷‍♂️ In that case I'd rather just throw an error earlier

Reply:
That also makes total sense. 

"they have to clean up"... do they ever? 😄